db/migrate/20190426011148_add_upload_foreign_keys_to_user_profiles.rb

```diff
@@ -0,0 +1,34 @@
+require 'migration/column_dropper'
+
+class AddUploadForeignKeysToUserProfiles < ActiveRecord::Migration[5.2]
+  def up
+    %i{profile_background card_background}.each do |column|
+      Migration::ColumnDropper.mark_readonly(:user_profiles, column)
+    end
+
+    add_column :user_profiles, :profile_background_upload_id, :integer, null: true
+    add_column :user_profiles, :card_background_upload_id, :integer, null: true
+    add_foreign_key :user_profiles, :uploads, column: :profile_background_upload_id
```

Comment:
The default behavior of `add_foreign_key` is to use `on_delete: :restrict`. That makes it impossible to delete the `upload` row until the foreign key is removed. I think maybe we would prefer `on_delete: :nullify`. What do you think?

Reply:
I guess it depends. The default behavior is perfectly fine if you want to make sure that you don't accidentally delete the upload. I think it's fine as it is.

Reply:
I don't feel too strongly either way, would be happy for it to stay as `restrict`. However, it might be annoying if you can't `Upload.find(123).destroy` any more 🤔

Reply:
Yea it should be restrict here. We shouldn't be able to delete the upload if it is still tied to a profile or card background.